src/algorithms/standard/minmax.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#include "MinMax.h"
+#include "essentiamath.h"
+#include <sstream>
+
+using namespace essentia;
+using namespace standard;
+
+const char* MinMax::name = "MinMax";
+const char* MinMax::category = "Standard";
+const char* MinMax::description = DOC("This algorithm calculates the minimum or maximum value of an array.");
+
+MinMax::OpType MinMax::typeFromString(const std::string& name) const {
+  if (name == "min") return MIN;
+  if (name == "max") return MAX;
+
+  throw EssentiaException("MinMax: Unknown operation: ", name);
+}
+
+void MinMax::compute() {
+
+  const std::vector<Real>& input = _input.get();
+
+  switch (_type) {
+
+  case MIN:
+    {
+      if (input.empty())
+        throw EssentiaException("trying to get min of empty array");
```

Comment:
the c++ implementation of mean does this check inside that function, I don't know if we should also move this check there?

Reply:
I agree that it would be good to move this check inside the `argmin` function so it would also cover other algorithms using the method.

Reply:
We still may want to know in which algorithm the error has happen, so we can check here for EssentiaException and then provide a more informative message ("MinMax: ...")

Reply:
It seems like in other places in essentia once you catch an exception you output it to stderr. In this case I don't think this would work, because the behaviour of min will be undefined on an empty array. So should we catch the exception and re-raise it with the algorithm name prepended?

Reply:
yes, we did this